test/test_torch.py

```diff
@@ -5993,15 +5993,12 @@ def test_diagonal_multidim(self, device, dtype):
         self.assertEqual(expected.shape, result.shape)
         self.assertEqual(expected, result)
 
-    def _test_trace(self, device, dtype, legacy):
+    @dtypes(*torch.testing.get_all_dtypes(include_complex=False, include_bool=False, include_half=False, include_bfloat16=False))
+    def test_trace(self, device, dtype):
         def test(shape):
             tensor = make_tensor(shape, device, dtype, low=-9, high=9)
             diag = tensor.diag()
```

Comment:
The diag = tensor.diag() line is bogus, can you remove it please? (Sorry, I was the one who added it but I am only noticing it now)